### PR TITLE
feat: add --steps flag with range support to logs command

### PR DESCRIPTION
**Pull request by Miyamura80**
## Summary
- Adds `--steps` flag to `desktest logs` supporting comma-separated step numbers and ranges (e.g. `--steps 3-7`, `--steps 1,3,5-8`)
- Upgrades `parse_steps()` in `codify.rs` to handle range syntax (`N-M`), benefiting both `logs` and `codify` commands
- Keeps backward-compatible `--step N` for single-step viewing (conflicts_with ensures mutual exclusivity)

## Test plan
- [x] `cargo build` compiles cleanly
- [x] All 7 `parse_steps` tests pass (4 new: range, mixed, single-element range, invalid range)
- [ ] Manual: `desktest logs artifacts/ --steps 3-7` shows steps 3 through 7
- [ ] Manual: `desktest logs artifacts/ --steps 1,3,5-8` shows the expected steps
- [ ] Manual: `desktest logs artifacts/ --step 3` still works as before

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/edison-watch/desktest/pull/33" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: src/logs.rs [hunk 1]
```diff
@@ -44,10 +44,10 @@ pub fn print_logs(artifacts_dir: &Path, brief: bool, step: Option<usize>) -> Res
 
     if brief {
         print_brief(&entries);
-    } else if let Some(n) = step {
-        let matching: Vec<_> = entries.iter().filter(|e| e.step == n).collect();
+    } else if let Some(ref filter) = step_filter {
+        let matching: Vec<_> = entries.iter().filter(|e| filter.contains(&e.step)).collect();
```
[the remark this answers is not in the document]
Comment (Miyamura80) [reply src/logs.rs]: **Valid** — Good catch. For large ranges the linear scan would be unnecessarily slow.

**Status: Addressed** — Fixed in commit 920a01d. Now converts the filter `Vec` to a `HashSet<usize>` before filtering, giving O(1) lookups per entry.

File: src/codify.rs [hunk 2]
```diff
@@ -213,16 +213,32 @@ pub fn generate_replay_script(
...
-        .map(|s| {
-            s.trim()
-                .parse::<usize>()
-                .map_err(|e| AppError::Config(format!("Invalid step number '{s}': {e}")))
-        })
-        .collect()
+    let mut result = Vec::new();
+    for part in steps_str.split(',') {
+        let part = part.trim();
+        if let Some((start_str, end_str)) = part.split_once('-') {
```
[the remark this answers is not in the document]
Comment (Miyamura80) [reply src/codify.rs]: **Valid** — Inputs like `-5` or `5-` would produce confusing errors about empty strings rather than pointing to the malformed range syntax.

**Status: Addressed** — Fixed in commit 920a01d. Added an explicit guard for empty start/end tokens with a clear error message (`"expected format N-M"`), plus tests for both leading and trailing dash edge cases.

**Comment by Miyamura80** [comment]
@greptileai
